extensions/amp-lightbox/1.0/component.js

```diff
@@ -150,35 +152,63 @@ function LightboxWithRef(
     };
   }, [visible, animationRef, onAfterCloseRef]);
 
+  const measureOverflowRef = useRef(null);
+  const [hasOverflow, setHasOverflow] = useState(scrollable);
+  // useLayoutEffect is used so ability to scroll on overflowing
+  // content is resolved before the component visually updates.
+  useLayoutEffect(() => {
+    const container = lightboxRef.current;
+    const content = measureOverflowRef.current;
+    if (!content || !container) {
+      return;
+    }
+    const win = toWin(content.ownerDocument.defaultView);
+    if (!win) {
+      return undefined;
+    }
+    const observer = new win.ResizeObserver(() => {
+      setHasOverflow(
+        scrollable &&
+          (content./* OK */ scrollHeight > container./* OK */ clientHeight ||
+            content./* OK */ scrollWidth > container./* OK */ clientWidth)
+      );
+    });
+    observer.observe(content);
+    observer.observe(container);
+    return () => observer.disconnect();
+  }, [mounted, scrollable, setHasOverflow]);
+
   return (
     mounted && (
       <ContainWrapper
         ref={lightboxRef}
-        size={true}
-        layout={true}
-        paint={true}
+        size
+        layout
+        paint
         part="lightbox"
-        contentStyle={
-          // Prefer style over class to override `ContainWrapper`'s overflow
-          scrollable && {
-            overflow: 'scroll',
-            overscrollBehavior: 'none',
-          }
-        }
-        wrapperClassName={`${classes.defaultStyles} ${classes.wrapper} ${
-          scrollable ? '' : classes.containScroll
-        }`}
+        contentClassName={objstr({
+          [classes.content]: true,
+          [classes.scrollable]: scrollable,
+        })}
+        wrapperClassName={classes.wrapper}
         role="dialog"
-        tabindex="0"
+        tabIndex="0"
         onKeyDown={(event) => {
-          if (event.key === Keys.ESCAPE) {
+          if (event.key === Keys.Escape) {
             setVisible(false);
           }
         }}
         {...rest}
       >
-        <CloseButton as={closeButtonAs} onClick={() => setVisible(false)} />
-        {children}
+        <div ref={measureOverflowRef}>
+          <CloseButton as={closeButtonAs} onClick={() => setVisible(false)} />
+          {children}
+        </div>
+        {!hasOverflow && (
```

Comment:
Hmm. I'd have a backdrop no matter what. It helps because if a lightbox is styled to be smaller than the whole viewport, it shows a shaded area that can also implement cancel-on-click-outside-the-lightbox feature.

Reply:
I think it's reasonable to consider styling the lightbox smaller than the whole viewport to be breaking the lightbox... That's what 0.1 does currently: [playground](https://playground.amp.dev/?format=websites&_gl=1*184k8jk*_ga*YW1wLTVrcDR1bGNQYlJRZ0VkTHVaT2JiRUE.#share=PCEtLQogICMjIEludHJvZHVjdGlvbgoKICBUaGUgW2BhbXAtbGlnaHRib3hgXSgvY29udGVudC9hbXAtZGV2L2RvY3VtZW50YXRpb24vY29tcG9uZW50cy9yZWZlcmVuY2UvYW1wLWxpZ2h0Ym94LXYwLjEubWQpIGNvbXBvbmVudCBhbGxvd3MgZm9yIGEg4oCcbGlnaHRib3jigJ0gb3Igc2ltaWxhciBleHBlcmllbmNlIC0gd2hlcmUgdXBvbiB1c2VyIGludGVyYWN0aW9uIGEgY29tcG9uZW50IGV4cGFuZHMgdG8gZmlsbCB0aGUgdmlld3BvcnQsIHVudGlsIGl0IGlzIGNsb3NlZCBhZ2FpbiBieSB0aGUgdXNlci4KLS0+PCEtLSAtLT4KPCFkb2N0eXBlIGh0bWw+CjxodG1sIOKaoSBsYW5nPSJlbiI+CjxoZWFkPgo8bWV0YSBjaGFyc2V0PSJ1dGYtOCI+CiAgPHRpdGxlPmFtcC1saWdodGJveDwvdGl0bGU+CiAgPHNjcmlwdCBhc3luYyBzcmM9Imh0dHBzOi8vY2RuLmFtcHByb2plY3Qub3JnL3YwLmpzIj48L3NjcmlwdD4KICA8IS0tICMjIFNldHVwIC0tPgogIDwhLS0gSW1wb3J0IHRoZSBgYW1wLWxpZ2h0Ym94YCBjb21wb25lbnQgaW4gdGhlIGhlYWRlciAtLT4KICA8c2NyaXB0IGFzeW5jIGN1c3RvbS1lbGVtZW50PSJhbXAtbGlnaHRib3giIHNyYz0iaHR0cHM6Ly9jZG4uYW1wcHJvamVjdC5vcmcvdjAvYW1wLWxpZ2h0Ym94LTAuMS5qcyI+PC9zY3JpcHQ+CiAgPCEtLSBZb3UgY2FuIGNvbWJpbmUgdGhlIGxpZ2h0Ym94IGNvbXBvbmVudCB3aXRoIGBhbXAtYmluZGAuIC0tPgogIDxzY3JpcHQgYXN5bmMgY3VzdG9tLWVsZW1lbnQ9ImFtcC1iaW5kIiBzcmM9Imh0dHBzOi8vY2RuLmFtcHByb2plY3Qub3JnL3YwL2FtcC1iaW5kLTAuMS5qcyI+PC9zY3JpcHQ+CiAgPGxpbmsgcmVsPSJjYW5vbmljYWwiIGhyZWY9Imh0dHBzOi8vYW1wLmRldi9kb2N1bWVudGF0aW9uL2V4YW1wbGVzL2NvbXBvbmVudHMvYW1wLWxpZ2h0Ym94L2luZGV4Lmh0bWwiPgogIDxtZXRhIG5hbWU9InZpZXdwb3J0IiBjb250ZW50PSJ3aWR0aD1kZXZpY2Utd2lkdGgiPgogIDxzdHlsZSBhbXAtYm9pbGVycGxhdGU+Ym9keXstd2Via2l0LWFuaW1hdGlvbjotYW1wLXN0YXJ0IDhzIHN0ZXBzKDEsZW5kKSAwcyAxIG5vcm1hbCBib3RoOy1tb3otYW5pbWF0aW9uOi1hbXAtc3RhcnQgOHMgc3RlcHMoMSxlbmQpIDBzIDEgbm9ybWFsIGJvdGg7LW1zLWFuaW1hdGlvbjotYW1wLXN0YXJ0IDhzIHN0ZXBzKDEsZW5kKSAwcyAxIG5vcm1hbCBib3RoO2FuaW1hdGlvbjotYW1wLXN0YXJ0IDhzIHN0ZXBzKDEsZW5kKSAwcyAxIG5vcm1hbCBib3RofUAtd2Via2l0LWtleWZyYW1lcyAtYW1wLXN0YXJ0e2Zyb217dmlzaWJpbGl0eTpoaWRkZW59dG97dmlzaWJpbGl0eTp2aXNpYmxlfX1ALW1vei1rZXlmcmFtZXMgLWFtcC1zdGFydHtmcm9te3Zpc2liaWxpdHk6aGlkZGVufXRve3Zpc2liaWxpdHk6dmlzaWJsZX19QC1tcy1rZXlmcmFtZXMgLWFtcC1zdGFydHtmcm9te3Zpc2liaWxpdHk6aGlkZGVufXRve3Zpc2liaWxpdHk6dmlzaWJsZX19QC1vLWtleWZyYW1lcyAtYW1wLXN0YXJ0e2Zyb217dmlzaWJpbGl0eTpoaWRkZW59dG97dmlzaWJpbGl0eTp2aXNpYmxlfX1Aa2V5ZnJhbWVzIC1hbXAtc3RhcnR7ZnJvbXt2aXNpYmlsaXR5OmhpZGRlbn10b3t2aXNpYmlsaXR5OnZpc2libGV9fTwvc3R5bGU+PG5vc2NyaXB0PjxzdHlsZSBhbXAtYm9pbGVycGxhdGU+Ym9keXstd2Via2l0LWFuaW1hdGlvbjpub25lOy1tb3otYW5pbWF0aW9uOm5vbmU7LW1zLWFuaW1hdGlvbjpub25lO2FuaW1hdGlvbjpub25lfTwvc3R5bGU+PC9ub3NjcmlwdD4KICA8c3R5bGUgYW1wLWN1c3RvbT4KICAgIC5saWdodGJveCB7CiAgICAgIGJhY2tncm91bmQ6IHJnYmEoMCwwLDAsMC44KTsKICAgICAgd2lkdGg6IDEwMCU7CiAgICAgIGhlaWdodDogMTAwJTsKICAgICAgcG9zaXRpb246IGFic29sdXRlOwogICAgICBkaXNwbGF5OiBmbGV4OwogICAgICBhbGlnbi1pdGVtczogY2VudGVyOwogICAgICBqdXN0aWZ5LWNvbnRlbnQ6IGNlbnRlcjsKICAgIH0KICAgIC5saWdodGJveCBoMSB7CiAgICAgIGNvbG9yOiB3aGl0ZTsKICAgIH0KICAgICNteS1saWdodGJveCB7CiAgICAgIGhlaWdodDogNTAlOwogICAgfQogIDwvc3R5bGU+CjwvaGVhZD4KPGJvZHk+CiAgPCEtLSAjIyBCYXNpYyB1c2FnZSAtLT4KICA8IS0tCiAgICBUaGUgYGFtcC1saWdodGJveGAgY29tcG9uZW50IGRlZmluZXMgdGhlIGNoaWxkIGVsZW1lbnRzIHRoYXQgd2lsbCBiZSBkaXNwbGF5ZWQgaW4gYSBmdWxsLXZpZXdwb3J0IG92ZXJsYXkuCiAgICBUbyBjbG9zZSB0aGUgbGlnaHRib3ggdmlhIGNsaWNrIG9yIHRhcCB1c2UgdGhlIGBvbmAgYXR0cmlidXRlIG9uIG9uZSBvciBtb3JlIGVsZW1lbnRzIGluc2lkZSB0aGUgbGlnaHRib3guIEluIHRoaXMgZXhhbXBsZSB0aGUgdXNlciBjYW4gY2xpY2sgYW55d2hlcmUgaW4gdGhlIGxpZ2h0Ym94IHRvIGNsb3NlIGl0LgoKICAgIFRoZSBsaWdoYm94IGlzIHNob3duIHdoZW4gdGhlIHVzZXIgdGFwcyBvciBjbGlja3Mgb24gYW4gZWxlbWVudCB3aXRoIGBvbmAgYXR0cmlidXRlIHRoYXQgdGFyZ2V0cyB0aGUgaWQgb2YgYW4gYGFtcC1saWdodGJveGAgZWxlbWVudC4KICAtLT4KICA8ZGl2PgogICAgPGFtcC1saWdodGJveCBpZD0ibXktbGlnaHRib3giIGxheW91dD0ibm9kaXNwbGF5Ij4KICAgICAgPGRpdiBjbGFzcz0ibGlnaHRib3giIG9uPSJ0YXA6bXktbGlnaHRib3guY2xvc2UiIHJvbGU9ImJ1dHRvbiIgdGFiaW5kZXg9IjAiPgogICAgICAgIDxoMT5IZWxsbyBXb3JsZCE8L2gxPgogICAgICA8L2Rpdj4KICAgIDwvYW1wLWxpZ2h0Ym94PgogICAgPGJ1dHRvbiBvbj0idGFwOm15LWxpZ2h0Ym94Ij4KICAgICAgT3BlbiBsaWdodGJveAogICAgPC9idXR0b24+CiAgPC9kaXY+CgogIDwhLS0KICAgIEZvciBzaG93aW5nIGltYWdlcyBpbiBhIGxpZ2h0Ym94LCBBTVAgYWxzbyBmZWF0dXJlcyB0aGUgW2FtcC1pbWFnZS1saWdodGJveF0oL2RvY3VtZW50YXRpb24vY29tcG9uZW50cy9yZWZlcmVuY2UvYW1wLWltYWdlLWxpZ2h0Ym94Lmh0bWwpIGNvbXBvbmVudCwgd2hpY2ggZG9lc24ndCBuZWVkIHRoZSBpbWFnZSB0byBzcGVjaWZpZWQgdHdpY2UuCiAgLS0+CgogIDwhLS0gIyMgQ29tYmluYXRpb24gd2l0aCBhbXAtYmluZCAtLT4KICA8IS0tIGBhbXAtbGlnaHRib3hgIGNhbiBiZSBvcGVuZWQgYXMgYSByZXN1bHQgb2YgYGFtcC1iaW5kYCBldmFsdWF0aW9ucyBieSBiaW5kaW5nIGEgYm9vbGVhbiB2YWx1ZSB0byB0aGUgYG9wZW5gIGF0dHJpYnV0ZS4gTm90ZSB0aGF0IHdlJ3JlIGFsc28gbGlzdGVuaW5nIHRvIHRoZSBgbGlnaHRib3hDbG9zZWAgZXZlbnQgdG8gcmVzZXQgdGhlIHN0YXRlIG9mIHRoZSB2YXJpYWJsZS4gLS0+CiAgPGRpdj4KICAgIDxhbXAtbGlnaHRib3ggaWQ9Im15LWJpbmRhYmxlLWxpZ2h0Ym94IiBbb3Blbl09InNob3dMaWdodGJveCIgbGF5b3V0PSJub2Rpc3BsYXkiIG9uPSJsaWdodGJveENsb3NlOkFNUC5zZXRTdGF0ZSh7c2hvd0xpZ2h0Ym94OiBmYWxzZX0pIj4KICAgICAgPGRpdiBjbGFzcz0ibGlnaHRib3giIHJvbGU9ImJ1dHRvbiIgdGFiaW5kZXg9IjAiIG9uPSJ0YXA6bXktYmluZGFibGUtbGlnaHRib3guY2xvc2UiPgogICAgICAgIDxoMT5IZWxsbyBXb3JsZCE8L2gxPgogICAgICA8L2Rpdj4KICAgIDwvYW1wLWxpZ2h0Ym94PgogICAgPGJ1dHRvbiBvbj0idGFwOkFNUC5zZXRTdGF0ZSh7c2hvd0xpZ2h0Ym94OiB0cnVlfSkiPgogICAgICBPcGVuIGxpZ2h0Ym94CiAgICA8L2J1dHRvbj4KICA8L2Rpdj4KPC9ib2R5PjwvaHRtbD4=)